### ISSUE TITLE
Someone is typing

### ISSUE THREAD

**Issue by DhairyaBahl** [DhairyaBahl/React-Messenger-App#15]
**Show "xyz" is typing**

When someone is typing then it should show that xyz person is typing. This "xyz" is the name which the user will enter in the prompt at the time of opening the website. Also suggest alignment to show this message. Discussion open for the issue.

**Comment by sagrawal9689** [comment]
Hey @DhairyaBahl I was thinking on how to do this. Can we do it like this, whenever the input state is changed we can push the name of the user in a collection (firebase collection that stores names of people typing). And remove the name in just 0.1 or 0.2 seconds. We can show the names of people typing at the end messages as we did for messages.
what do u say :confused:

**Comment by DhairyaBahl** [comment]
@sagrawal9689 I think you are in a right direction but

1.  I don't get your point to remove the name in just 0.1 seconds. Why are we doing that ? 

2. We have two options: first is we should just show the name of one person who is typing randomly like in whatsapp. second option is we should show names of every person who is typing like in discord. Either way will be perfect.

3. If you intend to work on this issue then I suggest you to try each one of the two ways and then we can choose wisely.

4. We will have to add a new variable in the collection (firebase collection). That's the only way I can think right now. So, don't worry and just go ahead.

Hope you find this helpful :smile:

**Comment by sagrawal9689** [comment]
@DhairyaBahl yeah 0.1 is too small.

I meant that once input state is changed we display that the user is typing, but we need to know weather user has stopped typing or not.
For doing this whenever input state is changed we check weather this user is present in Typing list or not, If it is already present we do nothing , if it is not present we insert the username in the list and set timer to remove the username from the list after 1sec.

So basically If a user is not typing for a max period of 1 sec his name will get removed from the Typing list.

I will try doing this. :smile:

**Comment by DhairyaBahl** [comment]
@sagrawal9689 It sounds perfect to me. Go ahead :fire:

**Comment by Hamza3226** [comment]
Hello, I am a GSSOC'21 participant and would like to contribute to this project.

**Comment by DhairyaBahl** [comment]
@Hamza3226 Someone is already assigned to this issue let me confirm from him if he is working on the issue or not ! if he is not working on this currently then I will definitely assign this to you.

@sagrawal9689 Are you still working on the issue ?

**Comment by subhangi2731** [comment]
@DhairyaBahl @abhijeet007rocks8 please assign me this issue
